File: python/python3_12_0/Python3/PythonParserBase.py
##
## Project      : a helper class to implement specific PEG grammar expressions in an ANTLR4 grammar
##
## Developed by : Robert Einhorn
##

## Related PEG grammar expressions:
## &e
## https://peps.python.org/pep-0617/#e-3
##
## !e
## https://peps.python.org/pep-0617/#e-4

from antlr4 import InputStream, Parser
from typing import TextIO
import sys

class PythonParserBase(Parser):
    def __init__(self, input: InputStream, output: TextIO = sys.stdout):
        super().__init__(input, output)

    def isEqualCurrentTokenText(self, tokenText: str) -> bool:
        return self.getCurrentToken().text == tokenText

    def isnotEqualCurrentTokenText(self, tokenText: str) -> bool:
        return not self.isEqualCurrentTokenText(tokenText) # for compatibility with the '!' logical operator in other languages

File: python/python3_12/Java/PythonParserBase.java
import org.antlr.v4.runtime.*;

public abstract class PythonParserBase extends Parser {
    protected PythonParserBase(TokenStream input) {
        super(input);
    }

    // https://docs.python.org/3/reference/lexical_analysis.html#soft-keywords
    public boolean isEqualToCurrentTokenText(String tokenText) {
        return this.getCurrentToken().getText().equals(tokenText);
    }

    public boolean isnotEqualToCurrentTokenText(String tokenText) {
        return !this.isEqualToCurrentTokenText(tokenText); // for compatibility with the Python 'not' logical operator
    }
}
